float64 y
float64 z
float64 y_dot
float64 z_dot